segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/StorageFactoryCreator.java

```diff
@@ -18,14 +18,17 @@
 public interface StorageFactoryCreator {
     /**
      * API to create a storage factory with given configuration.
-     * @param setup     Configuration for the factory.
-     * @param executor  The storage factory is expected to use this ExecutorService for execution of its tasks.
+     *
+     * @param storageFactoryInfo Properties of storage factory to create.
+     * @param setup              Configuration for the factory.
+     * @param executor           The storage factory is expected to use this ExecutorService for execution of its tasks.
      */
-    StorageFactory createFactory(ConfigSetup setup, ScheduledExecutorService executor);
+    StorageFactory createFactory(StorageFactoryInfo storageFactoryInfo, ConfigSetup setup, ScheduledExecutorService executor);
 
     /**
-     * The unique name for the storage factory.
-     * @return  Unique name for the storage factory.
+     * The properties of the available storage factories.
+     *
+     * @return The array of StorageFactoryInfo.
      */
-    String getName();
```

Comment:
Is it intentional to remove this `getName()` method? I guess it is fine as otherwise the build would fail...

Reply:
The interface has been changed.
The implementation is now supposed to return array of valid option via
``` java
public StorageFactoryInfo[] getStorageFactories() {
```